plugins/destination/postgresql/client/write.go

```diff
@@ -21,9 +19,35 @@ var cqStatusToPgStatus = map[schema.Status]pgtype.Status{
 	schema.Present:   pgtype.Present,
 }
 
+// Postgres best performance for CQ per our benchmark is achieved with batch that includes different tables
+// given we can't predict how many objects will be in each table.
+// This is why if we encounter an error we can't know which object/table caused and related the error to the object
+// we we need to find the error and exit/stop the sync.
+// Also, it is not advice to split the batch into different tables (for postgresql) as
+// this will mean increased number of goroutines and will mean more connections which PostgreSQL is not great with.
+// Also, I've benchmarked CopyFrom and batch and seems there is no significant difference in CQ case.
+// maybe in the future we can provide a flag to switch batching mechanism.
+func (c *Client) findErrorInBatch(ctx context.Context, items []*batchItem) error {
+	for _, item := range items {
+		_, err := c.conn.Exec(ctx, item.sql, item.arguments...)
```

Comment:
Are queries guaranteed to behave the same the second time we execute them? I'm concerned this can generated unrelated errors